src/models/datastore.js

```diff
@@ -136,6 +136,15 @@ class DataStore {
       return Promise.reject(new Error('put() cannot take an existing schema.'));
     }
 
+    var dataKeys = Object.keys(data);
+    var filteredKeys = dataKeys.filter(function (keys) {
```

Comment:
Why not just do `typeof data[''] !== 'undefined'`? Anything else is by definition non-zero length.